### PR TITLE
`no-guard`

### LINKED ISSUE #83: Rule: `no-guard`
In the next release of Effector, `sample` will be able to `filter` signals. Shall we add a rule that suggest replacing `guard`-s by `sample`-s?

### PR DESCRIPTION
**Pull request by igorkamyshev**
fixes #83 

- [x] imports
- [x] docs
- [x] config — `prefer-sample` (preset to enable `no-forward` and `no-guard`)

### CODE REVIEW

File: rules/no-guard/no-guard.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+sample({ clock: trigger, source: $data, filter: Boolean, target: reaction });
+```
+
+Nice bonus: `sample` is extendable. You can add transformation by `fn`.
+
+```ts
+// 👎 could be replaced
+guard({
+  clock: trigger,
+  soruce: $data.map((data) => data.length),
```
Comment (45m0d3u5) [inline rules/no-guard/no-guard.md:19]: typo in word. source*

File: rules/no-guard/no-guard.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  filter: Boolean,
+  target: reaction,
+});
+
+// 👍 makes sense
+sample({
+  clock: trigger,
+  soruce: $data,
```
Comment (45m0d3u5) [inline rules/no-guard/no-guard.md:27]: same

Comment (45m0d3u5) [reply rules/no-guard/no-guard.md:27]: Yeah, but I have no access) 

// Please make sure you have the correct access rights

Comment (igorkamyshev) [reply rules/no-guard/no-guard.md:27]: You have to fork repo, commit changes to your fork and send a PR.

Comment (45m0d3u5) [reply rules/no-guard/no-guard.md:27]: thanks) will do it now

Comment (45m0d3u5) [reply rules/no-guard/no-guard.md:27]: Done, waiting for your approve

**Pull request merged**